contrib/scalar/scalar.c

```diff
@@ -1005,6 +1005,51 @@ static void dir_file_stats(struct strbuf *buf, const char *path)
 				    e_stat.st_size);
 }
 
+static int count_files(char *path)
+{
+	DIR *dir = opendir(path);
+	struct dirent *e;
+	int count = 0;
+
+	if (!dir)
+		return 0;
+
+	while ((e = readdir(dir)) != NULL)
+		if (!is_dot_or_dotdot(e->d_name) && e->d_type == DT_REG)
+			count++;
+
+	return count;
```

Comment:
Please call `closedir(dir)` here.

Reply:
Good point! This applies to other `opendir` calls I've added.